lib/graphql/client.ts

```diff
@@ -37,4 +60,109 @@ export class GraphQLClient {
       throw new GraphQLError(serializeGraphQLError(error));
     }
   };
+
+  /**
+   * Return subscription client connection params
+   */
+  private getConnectionParams = (): ConnectionParams => ({
+    Authorization: `Bearer ${this.auth.tokenManager.token?.accessToken}`
+  });
+
+  /**
+   * Create a subscription client
+   */
+  private createSubscriptionClient = (): SubscriptionClient => {
+    if (!this.auth.tokenManager.token) {
+      throw new UserUnauthorizedError();
+    }
+
+    const WebSocket = typeof window === "undefined" ? ws : window.WebSocket;
+
+    return new SubscriptionClient(
+      this.subscriptionEndpoint,
+      {
+        lazy: true,
+        connectionParams: this.getConnectionParams
+      },
+      WebSocket
+    );
+  };
+
+  /**
+   * Handle disconnection:
+   *
+   * 1. Refresh access token
+   * 2. Destroy existing subscription client
+   * 3. Resubscribe all previously active subscriptions
+   */
+  private handleDisconnection = async (): Promise<void> => {
```

Comment:
Unfortunately, the `subscriptions-transport-ws` doesn't provide any helper or logic to manage reconnection, so this forced me to keep track of all the active subscriptions at all times so that we can "resubscribe" after a disconnection.

It adds a bit of overhead but it's not too bad so I still went ahead and did it.

Reply:
Makes sense, but if the token refresh fails we should throw an error which the user can handle then.

Reply:
Great point, see my below comment regarding that, changes are in this commit: 67de95da193ac9c4b86712c10132d47da7da22a6